OptimizelySDKCore/OptimizelySDKCore/Optimizely.m

```diff
@@ -186,27 +195,39 @@ - (OPTLYVariation *)variation:(NSString *)experimentKey
 #pragma mark Forced variation methods
 - (OPTLYVariation *)getForcedVariation:(nonnull NSString *)experimentKey
                                 userId:(nonnull NSString *)userId {
+    NSMutableDictionary<NSString *, NSString *> *inputValues = [[NSMutableDictionary alloc] initWithDictionary:@{
+                                                                                                                    OptimizelyNotificationsUserDictionaryUserIdKey:[self ObjectOrNull:userId],
+                                                                                                                    OptimizelyNotificationsUserDictionaryExperimentKey:[self ObjectOrNull:experimentKey]}];
+    if (![self validateStringInputs:inputValues logs:@{}]) {
+        return nil;
+    }
     return [self.config getForcedVariation:experimentKey
                                     userId:userId];
 }
 
 - (BOOL)setForcedVariation:(nonnull NSString *)experimentKey
                     userId:(nonnull NSString *)userId
               variationKey:(nullable NSString *)variationKey {
-    return [self.config setForcedVariation:experimentKey
+    NSMutableDictionary<NSString *, NSString *> *inputValues = [[NSMutableDictionary alloc] initWithDictionary:@{
+                                                                                                                    OptimizelyNotificationsUserDictionaryUserIdKey:[self ObjectOrNull:userId],
+                                                                                                                    OptimizelyNotificationsUserDictionaryExperimentKey:[self ObjectOrNull:experimentKey]}];
+    return [self validateStringInputs:inputValues logs:@{}] && [self.config setForcedVariation:experimentKey
                                     userId:userId
                               variationKey:variationKey];
 }
 
 #pragma mark - Feature Flag Methods
 
 - (BOOL)isFeatureEnabled:(NSString *)featureKey userId:(NSString *)userId attributes:(nullable NSDictionary<NSString *, NSObject *> *)attributes {
-    if ([userId getValidString] == nil) {
-        [self.logger logMessage:OPTLYLoggerMessagesFeatureDisabledUserIdInvalid withLevel:OptimizelyLogLevelError];
-        return false;
-    }
-    if ([featureKey getValidString] == nil) {
-        [self.logger logMessage:OPTLYLoggerMessagesFeatureDisabledFlagKeyInvalid withLevel:OptimizelyLogLevelError];
+    
+    NSMutableDictionary<NSString *, NSString *> *inputValues = [[NSMutableDictionary alloc] initWithDictionary:@{
```

Comment:
The key names not match between input values and logs:
OptimizelyNotificationsUserDictionaryExperimentKey
vs
OptimizelyNotificationsUserDictionaryFeatureKey

We need test cases for these errors.